app/src/main/java/com/pileproject/drive/programming/visual/activity/ProgrammingActivityBase.java

```diff
@@ -118,43 +129,37 @@ private void findViews() {
         mExecButton = (Button) findViewById(R.id.programming_execButton);
     }
 
-    private void moveToAnotherActivity() {
+    private void moveToExecutionActivity() {
         mSpaceManager.saveExecutionProgram();
         String address = MachinePreferences.get(getApplicationContext()).getMacAddress();
+
         // TODO this check does not work when dissolves paring
-        if (address == null) {
-            AlertDialog alertDialog = new AlertDialog.Builder(ProgrammingActivityBase.this).setTitle(R.string.error)
-                    .setMessage(R.string.deviceselect_didNotSelectDevice)
-                    .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
-                        @Override
-                        public void onClick(DialogInterface dialog, int which) {
-                            startActivity(getIntentToDeviceList());
-                        }
-                    })
-                    .create();
-            alertDialog.setCanceledOnTouchOutside(false);
-            alertDialog.show();
-        } else {
+        if (address != null || DeployUtils.isOnEmulator()) {
             Intent intent = getIntentToExecute();
             intent.putExtra("is_connected", mIsConnected);
-            startActivityForResult(intent, EXECUTE_PROGRAM);
+            startActivityForResult(intent, ACTIVITY_RESULT_EXECUTE_PROGRAM);
+            return ;
         }
+
+        new AlertDialogFragment.Builder(this)
+                .setRequestCode(DIALOG_REQUEST_CODE_DID_NOT_SELECT_DEVICE)
+                .setMessage(R.string.setting_bluetoothMachineSelect_didNotSelectDevice)
+                .setPositiveButtonLabel(android.R.string.ok)
+                .setCancelable(false)
+                .show();
     }
 
-    public void onFinishSelectDialog(String selectedText, boolean isSample) {
-        mSpaceManager.deleteAllBlocks(); // delete existing blocks
-        if (isSample) { // load a sample program
-            mSpaceManager.loadSampleProgram(selectedText);
-        } else { // load a user program
-            mSpaceManager.loadUserProgram(selectedText);
-        }
+    private void moveToTitleActivity() {
+        mSpaceManager.saveExecutionProgram();
+
+        // TODO: go to title activity
```

Comment:
The app do nothing when this button is pressed even before my pull request.
I didn't come up with a neat way to jump to the title, since the intent differs according
to the concrete title activity.
Can you give me the formal way?